### PR TITLE
Up the version to 0.14

### PR DESCRIPTION
**Pull request by jeongyoonlee**
## Proposed changes

Up the CausalML version to 0.14. Fix the readthedocs build failure.

## Types of changes

What types of changes does your code introduce to **CausalML**?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/uber/causalml/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/uber/causalml)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

N/A

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -3,6 +3,43 @@
 Changelog
 =========
 
+0.14.0 (July 2023)
+-----------------
+- CausalML surpassed `2MM downloads <https://pepy.tech/project/causalml>`_ on PyPI and `4,100 stars <https://github.com/uber/causalml/stargazers>`_ on GitHub. Thanks for choosing CausalML and supporting us on GitHub.
+- We have 7 new contributors: @darthtrevino, @ras44, @AbhishekVermaDH, @@joel-mcmurry, @AlxClt, @kklein, and @volico. Thanks for your contributions!
```
Comment (paullo0106) [inline docs/changelog.rst:9]: @jeongyoonlee seems like there is a redundant `@` for _joel-mcmurry_

File: docs/changelog.rst [hunk 2]
```diff
@@ -3,6 +3,43 @@
...
+
+Updates
+~~~~~~~~~~~~~
+- Fix the readthedocs build failure by @jeongyoonlee in https://github.com/uber/causalml/pull/545
+- Add pyproject.toml with basic build dependencies for PEP518 compliance by @darthtrevino in https://github.com/uber/causalml/pull/553
+- bump numpy==1.23.2 in environment-py38.yml #338 by @ras44 in https://github.com/uber/causalml/pull/550
```
Comment (paullo0106) [inline docs/changelog.rst:15]: minor: `Bump numpy from 1.20.3 to 1.23.2`

**Review by paullo0106 — APPROVED** [review]
LGTM with minor comments, thanks!

**Comment by huigangchen** [comment]
LGTM. Thank you!

**Pull request merged**